src/sentry/profiles/flamegraph.py

```diff
@@ -176,3 +189,172 @@ def extract_profile_ids() -> list[str]:
+                parameters=[
+                    Column("sentry_tags.value"),
+                    Function(
+                        "indexOf",
+                        parameters=[
+                            Column("sentry_tags.key"),
+                            "thread.id",
+                        ],
+                    ),
+                ],
+                alias="active_thread_id",
+            ),
+        ],
+        where=[
+            Condition(Column("project_id"), Op.EQ, project_id),
+            Condition(Column("timestamp"), Op.GTE, params["start"]),
+            Condition(Column("timestamp"), Op.LT, params["end"]),
+            Condition(Column("group"), Op.EQ, span_group),
+            Condition(Column("profiler_id"), Op.NEQ, ""),
+        ],
+        limit=Limit(100),
+    )
+    request = Request(
+        dataset=Dataset.SpansIndexed.value,
+        app_id="default",
+        query=query,
+        tenant_ids={
+            "referrer": Referrer.API_PROFILING_PROFILE_FLAMEGRAPH.value,
+            "organization_id": organization_id,
+        },
+    )
+    data = raw_snql_query(
+        request,
+        referrer=Referrer.API_PROFILING_PROFILE_FLAMEGRAPH.value,
+    )["data"]
+    spans: dict[str, list[IntervalMetadata]] = {}
+    for row in data:
+        if row["profiler_id"] in spans:
+            spans[row["profiler_id"]].append(
+                {
+                    "active_thread_id": row["active_thread_id"],
+                    "start": row["start_timestamp_precise"],
+                    "end": row["end_timestamp_precise"],
+                }
+            )
+        else:
+            spans[row["profiler_id"]] = [
+                {
+                    "active_thread_id": row["active_thread_id"],
+                    "start": row["start_timestamp_precise"],
+                    "end": row["end_timestamp_precise"],
+                }
+            ]
+    return spans
+
+
+class SpanMetadata(TypedDict):
+    profiler_id: list[IntervalMetadata]
+
+
+def get_chunk_snuba_conditions_from_spans_metadata(
+    spans: dict[str, list[IntervalMetadata]],
+) -> list[BooleanCondition | Condition]:
+    cond = []
+    for profiler_id, intervals in spans.items():
+        chunk_range_cond = []
+        for interval in intervals:
+            start = interval.get("start")
+            end = interval.get("end")
+            chunk_range_cond.append(
+                And(
+                    [
+                        Condition(Column("end_timestamp"), Op.GTE, start),
+                        Condition(Column("start_timestamp"), Op.LT, end),
+                    ],
+                )
+            )
+        cond.append(
+            And(
+                [
+                    Condition(Column("profiler_id"), Op.EQ, profiler_id),
+                    Or(chunk_range_cond) if len(chunk_range_cond) >= 2 else chunk_range_cond[0],
+                ]
+            )
+        )
+    return [Or(cond)] if len(cond) >= 2 else cond
```

Comment:
This should be okay for 100 spans but know that there is a maximum query size so if we were to raise the number of spans in the future, you may need to split the query into smaller blocks.

Reply:
yep, one more thing I want to do next, is to replace the fixed amount of 100 with an option as we did for the old flamegraph. Then I'll play with that value a bit to see how far can we stretch it before we have to split it as you suggest. 👍🏻